Assets/Scripts/Interactable.cs

```diff
@@ -18,18 +23,66 @@ void Start()
     }
 
     public void Init(){
-        this.myRenderer = GetComponent<SpriteRenderer>();
+        this.myRenderers = new List<SpriteRenderer>();
+        if (GetComponent<SpriteRenderer>() != null){
+            this.myRenderers.Add(GetComponent<SpriteRenderer>());
+        }
         this.selected = false;
+        this.cost = cost;
+        this.player = GameObject.FindGameObjectWithTag("Player").GetComponent<Player>();
+
+        this.failed = false;
+        this.waitTime = 0f;
+    }
+
+    private void LateUpdate(){
+        this.FailedPurchase();
     }
 
     public void Select(){
         this.selected = true;
-        this.myRenderer.material = selectedMaterial;
+        foreach (SpriteRenderer renderer in this.myRenderers){
+            renderer.material = selectedMaterial;
+        }
     }
 
     public void Deselect(){
         this.selected = false;
-        this.myRenderer.material = defaultMaterial;
+        foreach (SpriteRenderer renderer in this.myRenderers){
+            renderer.material = defaultMaterial;
+        }
+    }
+
+    private void FailedPurchase(){
+        if (this.failed == true){
+            this.waitTime += Time.deltaTime;
+            foreach (SpriteRenderer renderer in this.myRenderers){
+                renderer.material = failMaterial;
+            }
+            if (this.waitTime > 0.1f){
+                this.waitTime = 0.0f;
+                this.failed = false;
+                this.Deselect();
+            }
+        }
+    }
+
+    public void TriggerFailedPurchase(){
+        this.failed = true;
+    }
+
+    public bool CheckIfEnoughMoney(){
+        if (this.player.money >= this.cost){
+            return true;
+        }
+        else if (this.player.money < this.cost){
```

Comment:
I think this functions probably needs removing, where it's used in area door and weapon station it can just be replaced with `this.player.money >= this.cost`. I think `this.TriggerFailedPurchase();` can just be done in the classes that inherit interactable.